### DISCUSSION TITLE
istioctl: How to pass multi-line value via --set

### DISCUSSION THREAD

**Discussion by ghost** [istio/istio#47051] [Q&A]
Hello all,

I am trying to set pilot.jwksResolverExtraRootCA via the istioctl install command. I initially passed the path to the cert, only to discover that istio expects this value to be the actual CA cert contents.

I am unable to pass the value of the certificate via cat (e.g. istioctl install --set values.pilot.jwksResolverExtraRootCA=$(cat ca-certs.pem)), I receive an incorrect format, must be path=value error. 

Digging into the code, this happens when istioctl validates the values passed via --set:
https://github.com/istio/istio/blob/2c2f4a83f4bf3ed955c58f17a653b0eceda7780d/operator/pkg/manifest/shared.go#L361C20-L361C20

Is there a way to pass a multi-line value through istioctl, or is it best that I just modify the profile file that I'm passing when installing istio?

**Comment by zirain** [comment]
you should use istiooperator file for complex configuration.

**Reply by ghost** [reply]
I am using istiooperator, I was hoping for an easy way to insert the ca cert contents there without hardcoding the ca cert, so our CI/CD can inject it when doing deployments, as opposed to hardcoding the cert.

Is it possible to use a pre-existing secret name there, or does it have to be the actual contents of the ca pem?
